controllers/discovery/initDiscoveryUserData.js

```diff
@@ -137,7 +143,8 @@ const InitDiscoveryUserData = async (req, res) => {
         offset: page * limit,
         minimumKarmaScore: MINIMUM_KARMA_SCORE,
         admin_user_id: process.env.BETTER_ADMIN_ID,
-        excluded_users: excluded_users?.splice(0, MAX_EXCLUDED_USERS)
+        excluded_user_ids: excluded_user_ids?.splice(0, MAX_EXCLUDED_USERS),
+        excluded_user_names: excluded_user_names?.splice(0, MAX_EXCLUDED_USERS)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `splice` here directly modifies the `excluded_user_ids` and `excluded_user_names` arrays. This could lead to unexpected behavior if these arrays are used later in the code. Consider using `slice` instead, which returns a new array and does not modify the original one.
```diff
-         excluded_user_ids: excluded_user_ids?.splice(0, MAX_EXCLUDED_USERS),
-         excluded_user_names: excluded_user_names?.splice(0, MAX_EXCLUDED_USERS)
+         excluded_user_ids: excluded_user_ids?.slice(0, MAX_EXCLUDED_USERS),
+         excluded_user_names: excluded_user_names?.slice(0, MAX_EXCLUDED_USERS)
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->